Maps/ArchipelagoCampaign/WoL/ap_engine_of_destruction.SC2Map/MapScript.galaxy

```diff
@@ -739,6 +816,37 @@ bool gt_Init03Units_Func (bool testConds, bool runActions) {
         if (autoCC5E2357_var == null) { break; }
         UnitSetPropertyFixed(autoCC5E2357_var, c_unitPropEnergyPercent, 100.0);
     }
+    auto775B494E_g = UnitGroup("AP_Queen", c_playerAny, RegionEntireMap(), UnitFilter(0, 0, (1 << c_targetFilterMissile), (1 << (c_targetFilterDead - 32)) | (1 << (c_targetFilterHidden - 32))), 0);
+    auto775B494E_u = UnitGroupCount(auto775B494E_g, c_unitCountAll);
+    for (;; auto775B494E_u -= 1) {
+        auto775B494E_var = UnitGroupUnitFromEnd(auto775B494E_g, auto775B494E_u);
+        if (auto775B494E_var == null) { break; }
+        UnitSetPropertyFixed(auto775B494E_var, c_unitPropEnergyPercent, 100.0);
+    }
+    autoD9B4D754_g = UnitGroup("AP_SentryAiur", c_playerAny, RegionEntireMap(), UnitFilter(0, 0, (1 << c_targetFilterMissile), (1 << (c_targetFilterDead - 32)) | (1 << (c_targetFilterHidden - 32))), 0);
+    autoD9B4D754_u = UnitGroupCount(autoD9B4D754_g, c_unitCountAll);
+    for (;; autoD9B4D754_u -= 1) {
+        autoD9B4D754_var = UnitGroupUnitFromEnd(autoD9B4D754_g, autoD9B4D754_u);
+        if (autoD9B4D754_var == null) { break; }
+        UnitSetPropertyFixed(autoD9B4D754_var, c_unitPropEnergyPercent, 100.0);
+    }
+    if ((libABFE498B_gv_aP_Triggers_Option_overridePlayerRace == "Zerg")) {
+        UnitSetPropertyFixed(gv_odin, c_unitPropLifeRegen, 2.0);
+        UnitBehaviorAdd(gv_odin, "AP_ZergOdin", gv_odin, 1);
+    }
+    else {
+        if ((libABFE498B_gv_aP_Triggers_Option_overridePlayerRace == "Prot")) {
+            UnitSetPropertyFixed(gv_odin, c_unitPropLifeMax, 1000.0);
+            UnitSetPropertyFixed(gv_odin, c_unitPropLife, 1000.0);
+            UnitSetPropertyFixed(gv_odin, c_unitPropShieldsMax, 1500.0);
+            UnitSetPropertyFixed(gv_odin, c_unitPropShields, 1500.0);
+            UnitSetPropertyFixed(UnitFromId(4), c_unitPropLifeMax, 100.0);
+            UnitSetPropertyFixed(UnitFromId(4), c_unitPropLife, 100.0);
+            UnitSetPropertyFixed(UnitFromId(4), c_unitPropShieldsMax, 150.0);
+            UnitSetPropertyFixed(UnitFromId(4), c_unitPropShields, 150.0);

```

Comment:
See the comment for Media Blitz

Reply:
Fixed life issues with both this map and media blitz.